src/pages/resultsView/mutualExclusivity/MutualExclusivityTable.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react';
+import * as _ from "lodash";
+import LazyMobXTable, {Column} from "../../../shared/components/lazyMobXTable/LazyMobXTable";
+import {MutualExclusivity} from "../../../shared/model/MutualExclusivity";
+import {observer} from "mobx-react";
+import {observable} from "mobx";
+import {Badge} from 'react-bootstrap';
+
+export interface IMutualExclusivityTableProps {
+    columns?: MutualExclusivityTableColumnType[];
+    data: MutualExclusivity[];
+    initialSortColumn?:string;
+}
+
+export enum MutualExclusivityTableColumnType {
+    GENE_A,
+    GENE_B,
+    P_VALUE,
+    LOG_ODDS_RATIO,
+    ASSOCIATION
+}
+
+type MutualExclusivityTableColumn = Column<MutualExclusivity>&{order?:number, shouldExclude?:()=>boolean};
+
+export class MutualExclusivityTableComponent extends LazyMobXTable<MutualExclusivity> {
+}
+
+export function formatPValue(pValue: number): string {
+    return pValue < 0.001 ? "<0.001" : pValue.toFixed(3);
+}
+
+export function formatPValueWithStyle(pValue: number): JSX.Element {
+
+    let formattedPValue = <span>{formatPValue(pValue)}</span>;
+    if (pValue < 0.05) {
+        formattedPValue = <b>{formattedPValue}</b>;
+    }
+    return formattedPValue;
+}
+
+export function formatLogOddsRatio(logOddsRatio: number): string {
+
+    if (logOddsRatio < -3) {
+        return "<-3";
+    } else if (logOddsRatio > 3 || Number.isNaN(logOddsRatio)) {
+        return ">3";
+    }
+    return logOddsRatio.toFixed(3);
+}
+
+@observer
+export default class MutualExclusivityTable extends React.Component<IMutualExclusivityTableProps, {}> {
+
+    @observable protected _columns:{[columnEnum:number]:MutualExclusivityTableColumn};
+    @observable data: MutualExclusivity[];
+
+    constructor(props:IMutualExclusivityTableProps)
+    {
+        super(props);
+        this._columns = {};
+        this.data = props.data;
+        this.generateColumns();
+    }
+
+    public static defaultProps = {
+        columns: [
+            MutualExclusivityTableColumnType.GENE_A,
+            MutualExclusivityTableColumnType.GENE_B,
+            MutualExclusivityTableColumnType.P_VALUE,
+            MutualExclusivityTableColumnType.LOG_ODDS_RATIO,
+            MutualExclusivityTableColumnType.ASSOCIATION
+        ],
+        initialSortColumn: "p-Value"
+    };
+
+    public setData(data: MutualExclusivity[]) {
+        this.data = data;
```

Comment:
this strikes me as a React antipattern.  Any data used by this component should be passed in a prop.   Since data is already assigned to a prop value above, what is purpose of this?